### PR TITLE
Implementing time correlation functions

### PR DESCRIPTION
**Pull request by matteobecchi**
Requested Reviewers: @andrewtarzia

I thought it makes sense to implement this in dynsight because, although there are quick ways to compute the time correlation function for a signal using some numpy functions, this adds some useful features: 

* you give as input the signals for all the particles of a system, and the function "self_time_correlation" automatically returns the mean over all the particles; 
* the function "cross_time_correlation", instead, computes the cross correlations between the signals of different particles. 

Basically, we are adding the layer of "many body systems" over the time correlation function.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ maintainers = [
 ]
 
 dependencies = [
+  "numpy < 2",
```
Comment (andrewtarzia) [inline pyproject.toml:14]: Please leave a comment for why we are doing this and when we can remove this limitation

File: src/dynsight/_internal/analysis/time_correlations.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
```
Comment (andrewtarzia) [inline src/dynsight/_internal/analysis/time_correlations.py:1]: Why is this needed?

File: src/dynsight/_internal/analysis/time_correlations.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    the value at t = 0 is equal to 1.
+
+    * Author: Matteo Becchi
+
+    Parameters:
+        data:
+            Array of shape (n_particles, n_frames).
+
+        max_delay:
+            The TCF will be computed for all the delay values from zero to
```
Comment (andrewtarzia) [inline src/dynsight/_internal/analysis/time_correlations.py:27]: So, this is delay in frames, right? clarify.

**Review by andrewtarzia — APPROVED** [review]

**Comment by matteobecchi** [comment]
meme_PR

**Review by andrewtarzia — APPROVED** [review]
